### PR TITLE
fix: Test `Lexer` and fix bugs found

### PR DESCRIPTION
**Pull request by herluf-ba**
This PR adds tests for the new `Lexer`. It specifically tests each error production of Lexer.
It also adds fixes for the errors that testing discovered.

Suggestions for additional tests are very welcome!

### CODE REVIEW

**Comment by magnus-madsen** [comment]
This seems to do more than adding some tests?

File: main/test/ca/uwaterloo/flix/language/phase/TestLexer.scala [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+       """.stripMargin
+    val result = compile(input, Options.TestWithLibNix)
+    expectError[LexerError.UnterminatedChar](result)
+  }
+
+  test("LexerError.UnterminatedInfixFunction.01") {
+    val input =
+      s"""
+         |def f(): Int32 = 1 `add 2
+       """.stripMargin
```
Comment (magnus-madsen) [inline main/test/ca/uwaterloo/flix/language/phase/TestLexer.scala:198]: Notice that you have a new line in all these tests.

Maybe you want to double the tests s.t. you also check for un-terminated XYZ when the file ends on the same line.

**Comment by herluf-ba** [comment]
Yes, It also fixes the errors I discovered during testing. I will change the title to reflect that.

**Comment by magnus-madsen** [comment]
Performance impact? Check with --Xbenchmark-phases.

File: main/test/ca/uwaterloo/flix/language/phase/TestLexer.scala [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+    val input =
+      s"""
+         |def f(): Int32 = 1 `add/*this is a block comment*/` 2
+       """.stripMargin
+    val result = compile(input, Options.TestWithLibNix)
+    expectError[LexerError.UnterminatedInfixFunction](result)
+  }
+
+  test("LexerError.UnterminatedInfixFunction.03") {
+    val input =
```
Comment (magnus-madsen) [inline main/test/ca/uwaterloo/flix/language/phase/TestLexer.scala:220]: Meh. Why so ugly? It could simply be on one line now. val input = "myStringHere".

Comment (herluf-ba) [reply main/test/ca/uwaterloo/flix/language/phase/TestLexer.scala:220]: You mean:
```scala
val input = s"1 `add 2"
```
I was just trying to be consistent with the formatting I found in the other test files. I will convert to onelines where possible.

**Comment by herluf-ba** [comment]
There seems to be no significant performance impact
```
Lexer                          143,6 ms (01,4%))
...
Parser                         560,2 ms (05,4%)
```

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**